File: Tool/PreUtil/src/output/javascript/easy/BalancedBinaryTree.js
/**https://leetcode.com/problems/balanced-binary-tree/ */
//Given a binary tree, determine if it is height-balanced.


/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isBalanced = function(root) {
    if(root == null){
        return true;
    }
    return Math.abs(height(root.left) - height(root.right)) <= 1 && isBalanced(root.left) && isBalanced(root.right);
    
    
};

    

File: Tool/PreUtil/src/output/c/easy/BalancedBinaryTree.c
/**https://leetcode.com/problems/balanced-binary-tree/ */
//Given a binary tree, determine if it is height-balanced.

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool isBalanced(struct TreeNode* root){
    if(root == NULL){
        return true;
    }
    int left = maxDepth(root->left);
    int right = maxDepth(root->right);
    if(abs(left - right) > 1){
        return false;
    }
    return isBalanced(root->left) && isBalanced(root->right);

}
    